easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -107,6 +108,14 @@ def configure_step(self, srcdir=None, builddir=None):
         # show what CMake is doing by default
         options.append('-DCMAKE_VERBOSE_MAKEFILE=ON')
 
+        # don't pick up on system Boost if Boost is included as dependency
+        # - specify Boost location via -DBOOST_ROOT
+        # - instruct CMake to not search for Boost headers/libraries in other places
+        # - disable search for Boost CMake package configuration file
+        boost_root = get_software_root('Boost')
+        if boost_root:
```

Comment:
Add an option `enable_system_boost` with a default of false so this can be switched off in a corner case.

Also should we care that this may only be relevant to versions of CMake that supported this functionality?

Reply:
Looks like `Find_Boost` goes back to 2.8.4 (oldest in EB) so should be fine